overlord/snapstate/backend/snapdata.go

```diff
@@ -108,9 +123,37 @@ func copySnapData(oldSnap, newSnap *snap.Info) (err error) {
 	return nil
 }
 
+func trashPath(path string) string {
+	return path + ".old"
+}
+
+// trash moves path aside
+func trash(path string) error {
+	return os.Rename(path, trashPath(path))
+}
+
+// untrash moves the trash for path back in
+func untrash(path string) error {
+	return os.Rename(trashPath(path), path)
+}
+
+// clearTrash removes the trash made for path
+func clearTrash(path string) error {
+	return os.RemoveAll(trashPath(path))
+}
+
 // Lowlevel copy the snap data (but never override existing data)
 func copySnapDataDirectory(oldPath, newPath string) (err error) {
 	if _, err := os.Stat(oldPath); err == nil {
+		if _, err := os.Stat(newPath); err == nil {
+			// newPath already exists; back up
+			if err := trash(newPath); err != nil {
```

Comment:
Or perhaps here is a better place to handle that case. If the oldPath exists, the newPath exists, and the trash exists, that's probably enough data to allow us to kill an ancient broken trash.